x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_confirmation_modal.test.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import '../../../../__mocks__/kea.mock';
+import { setMockActions, setMockValues } from '../../../../__mocks__';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { shallow } from 'enzyme';
+
+import { LogRetentionConfirmationModal } from './log_retention_confirmation_modal';
+import { ELogRetentionOptions } from './types';
+import { GenericConfirmationModal } from './generic_confirmation_modal';
+
+describe('<LogRetentionConfirmationModal />', () => {
+  beforeAll(() => {
+    // LogRetentionConfirmationModal contains EuiModals, which utilize React Portals,
+    // so we must mock `createPortal` to get the rendered element directly,
+    // instead of letting it be placed normally elsewhere in DOM (outside of jest's domain)
+    // @ts-ignore
+    ReactDOM.createPortal = jest.fn((element) => {
+      return element;
+    });
+  });
+
+  const actions = {
+    closeModals: jest.fn(),
+    saveLogRetention: jest.fn(),
+  };
+
+  const values = {
+    openedModal: null,
+    logRetention: {
+      analytics: {
+        enabled: true,
+        retentionPolicy: {
+          isDefault: true,
+          minAgeDays: 180,
+        },
+      },
+      api: {
+        enabled: true,
+        retentionPolicy: {
+          isDefault: true,
+          minAgeDays: 7,
+        },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockActions(actions);
+    setMockValues(values);
+  });
+
+  afterEach(() => {
+    // Remove the jest mock on createPortal
+    // @ts-ignore
+    ReactDOM.createPortal.mockClear();
```

Comment:
why is this ts-ignore and comment here? Do we need it, or can we do

```suggestion
    (ReactDOM.createPortal as jest.Mock).mockClear();
```

or just `jest.clearAllMocks` 🤷 
edit: which we're already doing in beforeEach - can we lose this block entirely?

Reply:
Deleted per previous comment.

Reply:
https://github.com/elastic/kibana/pull/83009/commits/99463bb6ab73cc0db72df7b445e5ab669e55380e